domain/user/types.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package user
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/juju/utils/v3"
+)
+
+type User struct {
+	// Id is the unique identifier for the user.
+	Id UUID
+
+	// CreatedAt is the time that the user was created at.
+	CreatedAt time.Time
+
+	// DisplayName is a user-friendly name represent the user as.
+	DisplayName string
+
+	// Name is the username of the user.
+	Name string
+
+	// CreatorUUID is the associated user that created this user.
+	CreatorUUID UUID
+}
+
+type UUID string
+
+func NewUUID() (UUID, error) {
+	uuid, err := utils.NewUUID()
+	if err != nil {
+		return UUID(""), err
+	}
+	return UUID(uuid.String()), nil
+}
+
+func (u UUID) Validate() error {
```

Comment:
This doesn't appear to be used.

Reply:
Thanks, I forgot to add it... now it used by all the functions (where UUID is a param)